cmd/migrateblobs/main.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// This command migrates blobstore blobs of charms and resources for all
+// entities.  This command is intended to be run on the production db and then
+// discarded.  The first time this command is executed, all the entities are
+// updated.  Subsequent runs migrate missing destination blobs.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/juju/loggo"
+	"github.com/juju/utils/parallel"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/goose.v2/client"
+	"gopkg.in/goose.v2/errors"
+	"gopkg.in/goose.v2/identity"
+	"gopkg.in/goose.v2/swift"
+	"gopkg.in/mgo.v2"
+
+	"gopkg.in/juju/charmstore.v5-unstable/config"
+)
+
+var (
+	logger        = loggo.GetLogger("migrateblobs")
+	loggingConfig = flag.String("logging-config", "INFO", "specify log levels for modules e.g. <root>=TRACE")
+)
+
+const maxRetries = 10
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "usage: %s [options] <config path> \n", filepath.Base(os.Args[0]))
```

Comment:
honestly, I'd just say "usage: migrateblobs ...." rather than bothering with the Base shenanigans.